operator/operatortest/envtest.go

```diff
@@ -181,6 +181,31 @@ func (ts *Suite) FetchResources(objectList client.ObjectList, opts ...client.Lis
 	ts.Require().NoError(ts.Client.List(ts.Context, objectList, opts...))
 }
 
+// ExistsResource checks if the given resource exists.
+// Test fails for any error.
+func (ts *Suite) ExistsResource(name types.NamespacedName, object client.Object) {
+	ts.Require().NoError(ts.Client.Get(ts.Context, name, object))
+}
+
+// NotFoundResource checks if the given resource is missing.
+// Test fails if the resource exists or there's another error.
+func (ts *Suite) NotFoundResource(name types.NamespacedName, object client.Object) {
+	err := ts.Client.Get(ts.Context, name, object)
+	if err != nil {
+		ts.Require().True(apierrors.IsNotFound(err))
+	} else {
+		ts.Require().NotEmpty(object.GetDeletionTimestamp())
+	}
+}
+
+// NoFinalizers checks if the given resource has finalizers.
+// Test fails if the resource has any finalizers.
+func (ts *Suite) NoFinalizers(name types.NamespacedName, object client.Object) {
+	err := ts.Client.Get(ts.Context, name, object)
+	ts.Require().NoError(err)
+	ts.Require().Empty(object.GetFinalizers())
+}
```

Comment:
Why do we need a function for this? Why not just doing
```go
obj := &MyObject{ meta etc...}
ts.FetchResource(..., obj)
ts.Assert().Empty(object.GetFinalizers())
```
in the test code?

I mean chances are high that the testing code needs to fetch an object in any case and do a number of assertions, no need to GET it for every kind of assertion, that is just slow.

Reply:
Well the function clearly states what it is doing. Also I followed the style that is already present in _envtest_. Having another Get from the suit that is in memory should not pose a threat to performance, and besides, It's easier to read the assertion by removing the logic behind a function.

Reply:
> Having another Get from the test suit that is in memory should not pose a threat to performance

That's wrong. The Kubernetes cluster in envtest is _another process_ and envtest accesses it through a network port in `localhost`. So there is technically network call in between that we can avoid.

> Well the function clearly states what it is doing.

I'm not questioning the purpose or intent, I question whether it's worth having asserting functions for some properties in an object spec. By that logic we would also need `HasFinalizers`, `HasLabels`, `NotHasHabels` methods etc.


> Also I followed the style that is already present in _envtest_.

Not quite. Currently, there are no assertion functions in envtest.go... It only provides some helpers method interacting with Kubernetes but only with some error assertions, not property verification.

Reply:
>Having another Get from the test suit that is in memory should not pose a threat to performance
Well I didn't know we are that concerned about the test performance. Maybe we should write some guides what we appreciate more in a test suite, readability or performance.

>I'm not questioning the purpose or intent, I question whether it's worth having asserting functions for some >properties in an object spec. By that logic we would also need HasFinalizers, HasLabels, NotHasHabels methods etc.
`AssertResourceNotExists` this method does it as well. What is wrong with having such methods in a test suite? For me it helps with readability.

>Not quite. Currently, there are no property assertion functions in envtest.go... It only provides some helpers method >interacting with Kubernetes but only with some error assertions, not property verification.
Well we could move these methods inside the test file if that bothers you.